stockfish/models.py

```diff
@@ -677,25 +690,12 @@ def get_wdl_stats(self) -> Optional[List[int]]:
             )
 
         self._go()
-        lines: List[List[str]] = []
-        while True:
-            text = self._read_line()
-            splitted_text = text.split(" ")
-            lines.append(splitted_text)
-            if splitted_text[0] == "bestmove":
-                break
-        for current_line in reversed(lines):
-            if current_line[0] == "bestmove" and current_line[1] == "(none)":
-                return None
-            elif "multipv" in current_line:
-                index_of_multipv = current_line.index("multipv")
-                if current_line[index_of_multipv + 1] == "1" and "wdl" in current_line:
-                    index_of_wdl = current_line.index("wdl")
-                    wdl_stats: List[int] = []
-                    for i in range(1, 4):
-                        wdl_stats.append(int(current_line[index_of_wdl + i]))
-                    return wdl_stats
-        raise RuntimeError("Reached the end of the get_wdl_stats function.")
+        lines = list(reversed(self._get_sf_go_command_output()))
+        if lines[0].startswith("bestmove (none)"):
+            return None
+        split_line = next(line.split(" ") for line in lines if " multipv 1 " in line)
```

Comment:
```suggestion
        lines = self._get_sf_go_command_output()
        if lines[-1].startswith("bestmove (none)"):
            return None
        split_line = [line.split(" ") for line in lines if " multipv 1 " in line][-1]
```
Since we are now not looping through the list, but taking specific elements directly, we would theoretically not have to reverse it (same for `get_evaluation`). I would find it a bit easier to read that way. But it's your choice, if you want to leave it that way, feel free.

Reply:
@kieferro Thx for the feedback! I agree this would make it more readable, since for just one element it's less steps conceptually.